src/log.c

```diff
@@ -353,11 +322,19 @@ int LogReset(Log *log, raft_index_t index, raft_term_t term)
 int LogLoadEntries(Log *log)
 {
     log->num_entries = 0;
+    long calc_crc = 0;
 
     if (readHeader(log) != RR_OK) {
         return RR_ERROR;
     }
 
+    /* calculate crc to header end */
```

Comment:
so, I'm not sure of this.  

1) readHeader (for some reasn) might be called in other situations, much like readEntry()
2) since readEntry can't update it itself (because it can be called randomly), I'm not sure readHeader() should, even if it could.

I'm unconvinced readHeader() should even do the validation itself, but it was the cleanest way to get it into the code, without modifying other locations.

Reply:
I think we should treat the crc and the entry data as a single entity. I wonder about the validation - I think we can have it optional in the read itself, or have a new RR entry type that contains also the crc, and only read it in the read func, but do a validation after, only when needed. WDYT?

Reply:
ok, changed